### DISCUSSION TITLE
Add config_path to Lutris/Xemu runner's parameters

### DISCUSSION THREAD

**Discussion by Bug-Fighter** [lutris/lutris#5230] [Ideas]
Lutris already take into account two Xemu command line arguments: -full-screen and -dvd_path <iso>. In order to create a dedicated sandbox for each game running with Lutris/Xemu, could you add the Xemu argument -config_path <config> to Lutris/Xemu runner's parameters. 

In other words, within YAML script, a dedicated config file can be created using the write_config command. In that way, a specific hard disk image can be choosen, and so, each game will run in a private space. Xemu general settings can also be tuned to game's requirements.

    

> write_config:

      description: Xemu configuration file
      file: $GAMEDIR/xemu.toml
      merge: false
      data:
        general:
          show_welcome: "false"
          last_viewed_menu_index: 6
        input.bindings:
          port1: "'keyboard'"
        sys.files:
          bootrom_path: "'$GAMEDIR/mcpx_1.0.bin'"
          flashrom_path: "'$GAMEDIR/Complex_4627.bin'"
          eeprom_path: "'$HOME/.local/share/xemu/xemu/eeprom.bin'"
          hdd_path: "'$GAMEDIR/xbox_hdd.qcow2'"
          dvd_path: "'$GAMEDIR/DaVinciCode.xiso.iso'"

For the moment, the created config file, xemu.toml, is merged into the Xemu config folder : ~/.local/share/xemu/xemu. Using the Xemu command line argument config_path seems to be a more suitable process.

Xemu command line argument -snapshot may be also taken into account as it discards any writes to the HDD image after exit. It could be another step to protect data.

Don't worry, I will not ask for adding all Qemu's inherited command line arguments !

Best regards

**Comment by Bug-Fighter** [comment]
So here I am once more !!!

Another way to obtain such a result inexpensively would be to systematically assign the args parameter to all runners within the game section of the Lutris' scripts.

That's already done for some of them:

- wine :  gives arguments to the running game
- dosbox, scummvm, zdoom, MAME : gives arguments to the runner

It can be added to (as an example):

- ryujinx : args: --fullscreen $GAMEDIR/ROM.nsp 
- HATARI : args: --configfile <filename> --language <x> 
-  melonDS : args: --fullscreen --archive-file ROM.nds
-  xemu : args: -config_path $GAMEDIR/xemu.toml

The args parameter would be an easy way to fine-tune runners.

See you next time

**Comment by strycore** [comment]
>  In order to create a dedicated sandbox for each game running with Lutris/Xemu

Ok what's up with that? Why would you run each Xbox game in it's own sandbox. I have an initial setup for Xemu's hard drive and it works fine for all games.

**Comment by Bug-Fighter** [comment]
Regarding Xemu, I want to keep an archive containing all games data, the xiso file of course, and all personal saves and parameters of that game without any other one. So the only way I found to do this easily was to use a dedicated hard drive image. This idea may seem preposterous and I go with it. 

I fully agree that Lutris plays well with Xemu, it's the reason why I use it. 

Regarding Lutris, this curious idea reveals a more general topic as I wrote in a second comment. In order to allow every user to fine tune there runners according to their smart , or foolish, requests, would it be possible to make the args parameter a generic one in the game section of the YAML script. 
In one hand, that's an  easy way to take advantage of every runners strength without writing too many code. In the other hand, it could lower lutris' control on the runners.

As a conclusion, I'm only discussing, not requesting. I hope this input gives you a snapshot of some users' way of thinking.
